checks/file-checks.php

```diff
@@ -35,6 +35,29 @@ function check_phar() {
 		}
 	}
 
+	function check_application() {
+		$application_files = [ '.a', '.bin', '.bpk', '.deploy', '.dist', '.distz', '.dmg', '.dms', '.DS_Store', '.dump', '.elc', '.exe', '.iso', '.lha', '.lrf', '.lzh', '.o', '.obj', '.phar', '.pkg', '.sh', '.so' ];
+
+		$files = array_filter( $this->files, function( $file ) use ( $application_files ) {
+			$extension = sprintf( '.%s', pathinfo( $file, PATHINFO_EXTENSION ) );
+			return in_array( $extension, $application_files, true );
+		} );
+
+		if ( $files ) {
+			$notice_or_error = ( ( defined( 'WP_DEBUG' ) && WP_DEBUG ) || 'production' !== wp_get_environment_type() ) ? Notice::class : Error::class;
```

Comment:
It would be nice if we could leverage this in other checks, so it's consistent across the board.

Maybe we should move this to a method on the `abstract` so that we can do `return $this->throw();` And it determines if a `Notice` or `Error` based on Env or Debug.

Reply:
Done in [Abstract throw_error](https://github.com/WordPress/plugin-check/pull/43/commits/02addaa6d94654adbe66ed925e71fd4bd1369234). I'm not totally sold on the name `throw_notice()` maybe we just stick to `throw()` like you suggested.